flink-runtime/src/main/java/org/apache/flink/runtime/state/filesystem/FsSegmentDataInputStream.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state.filesystem;
+
+import org.apache.flink.core.fs.FSDataInputStream;
+import org.apache.flink.core.fs.FSDataInputStreamWrapper;
+import org.apache.flink.core.fs.WrappingProxyCloseable;
+
+import java.io.IOException;
+
+/**
+ * This class is a {@link org.apache.flink.util.WrappingProxy} for {@link FSDataInputStream} that is
+ * used to read from a file segment. It is opened with a starting position of the file. It treats
+ * the argument of seek(long) as an offset relative to the starting position, rather than an
+ * absolute value.
+ */
+public class FsSegmentDataInputStream extends FSDataInputStreamWrapper
+        implements WrappingProxyCloseable<FSDataInputStream> {
+
+    private final long startingPosition;
+
+    private long endingPosition;
+
+    private long markPosition;
+
+    private long curPosInFile;
+
```

Comment:
I'm wondering if we could let this class simply proxy operations to the `FSDataInputStreamWrapper` and omit these members?

Reply:
Sure. This can simplify the code for ```FsSegmentDataInputStream```.

Note that, not all implementations of ```FSDataInputStream``` supports mark/reset. Therefore ```FsSegmentDataInputStream#markSupported()``` does not necessary return 'true' now.